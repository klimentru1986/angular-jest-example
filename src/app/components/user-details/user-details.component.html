<div *ngIf="userInfo">
  <h4>Данные пользователя</h4>
  <div>Идентификатор: {{userInfo.id}}</div>
  <div>Имя: {{userInfo.name}}</div>
  <div>Возраст: {{userInfo.age}}</div>
</div>