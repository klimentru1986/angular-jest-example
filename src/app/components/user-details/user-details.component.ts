import { Component, OnInit, Input } from '@angular/core';
import { User } from '../../models/user.model';

@Component({
  selector: 'app-user-details',
  templateUrl: './user-details.component.html',
  styleUrls: ['./user-details.component.css']
})
export class UserDetailsComponent implements OnInit {
  @Input() userInfo: User;

  constructor() {}

  ngOnInit() {}
}
