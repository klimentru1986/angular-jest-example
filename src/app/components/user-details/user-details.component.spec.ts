import { ComponentFixture, TestBed } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { UserDetailsComponent } from './user-details.component';

describe('UserDetailsComponent', () => {
  let comp: UserDetailsComponent;
  let fixture: ComponentFixture<UserDetailsComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [UserDetailsComponent],
      schemas: [NO_ERRORS_SCHEMA]
    });
    fixture = TestBed.createComponent(UserDetailsComponent);
    comp = fixture.componentInstance;
    comp.userInfo = {
      id: 'qwe123',
      name: 'Иванов Иван Иванович',
      age: 33
    };
  });

  it('can load instance', () => {
    // Assert
    expect(comp).toBeTruthy();
  });

  it('should renders correctly', () => {
    // Act
    fixture.detectChanges();
    const elHtml = fixture.nativeElement;

    // Assert
    expect(elHtml).toMatchSnapshot();
  });
});
