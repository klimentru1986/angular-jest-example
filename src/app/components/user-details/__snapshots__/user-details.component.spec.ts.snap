// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetailsComponent should renders correctly 1`] = `
<div
  id="root1"
  ng-version="5.2.5"
>
  
  <div>
    
  
    <h4>
      Данные пользователя
    </h4>
    
  
    <div>
      Идентификатор: qwe123
    </div>
    
  
    <div>
      Имя: Иванов Иван Иванович
    </div>
    
  
    <div>
      Возраст: 33
    </div>
    

  </div>
</div>
`;
